internal/provider/storage/storage_quota_rules_resource.go

```diff
@@ -0,0 +1,423 @@
+						Optional:            true,
+						Computed:            true,
+						PlanModifiers: []planmodifier.Int64{
+							int64planmodifier.UseStateForUnknown(),
+						},
+					},
+				},
+			},
+			"id": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+		},
+	}
+}
+
+// Configure adds the provider configured client to the resource.
+func (r *StorageQuotaRulesResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+	config, ok := req.ProviderData.(connection.Config)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected Config, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+	}
+	r.config.ProviderConfig = config
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (r *StorageQuotaRulesResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data StorageQuotaRulesResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	errorHandler := utils.NewErrorHandler(ctx, &resp.Diagnostics)
+	// we need to defer setting the client until we can read the connection profile name
+	client, err := connection.GetRestClient(errorHandler, r.config, data.CxProfileName)
+	if err != nil {
+		// error reporting done inside NewClient
+		return
+	}
+
+	var restInfo *interfaces.StorageQuotaRulesGetDataModelONTAP
+	if data.ID.ValueString() != "" {
+		restInfo, err = interfaces.GetStorageQuotaRulesByUUID(errorHandler, *client, data.ID.ValueString())
+		if err != nil {
+			// error reporting done inside GetStorageQuotaRulesByUUID
+			return
+		}
+	} else {
+		restInfo, err = interfaces.GetStorageQuotaRules(errorHandler, *client, data.Volume.Name.ValueString(), data.SVM.Name.ValueString(), data.Type.ValueString(), data.Qtree.Name.ValueString())
+		if err != nil {
+			// error reporting done inside GetStorageQuotaRules
+			return
+		}
+		data.Type = types.StringValue(restInfo.Type)
+		data.SVM.Name = types.StringValue(restInfo.SVM.Name)
+		data.Volume.Name = types.StringValue(restInfo.Volume.Name)
+		data.Qtree.Name = types.StringValue(restInfo.Qtree.Name)
+	}
+
+	// Files
+	elementTypes := map[string]attr.Type{
+		"hard_limit": types.Int64Type,
+		"soft_limit": types.Int64Type,
+	}
+	elements := map[string]attr.Value{
+		"hard_limit": types.Int64Value(restInfo.Files.HardLimit),
+		"soft_limit": types.Int64Value(restInfo.Files.SoftLimit),
+	}
+	objectValue, diags := types.ObjectValue(elementTypes, elements)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+	}
+	data.Files = objectValue
+	data.ID = types.StringValue(restInfo.UUID)
+
```

Comment:
Is 'user' missing?

Reply:
Nope, I tested it again, its good, successfully creating, modifying, deleting and updating the state file

Reply:
Is user not returned then?

Reply:
user can be added only through input, if not, its always empty and its not modifiable